File: 백준/Bronze/18312. 시각/시각.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, k] = input[0].split(' ');

const to_string2 = (a) => {
  return a / 10 < 1 ? '0' + a : a + '';
}

const check_sec = (sec, k) => {
  let temp = sec;
  let h = Math.floor(temp / (60 * 60));
  temp %= 60 * 60;
  let m = Math.floor(temp / 60);
  temp %= 60;
  
  let s = to_string2(h) + to_string2(m) + to_string2(temp);
  
  return s.includes(k)
}

const answer = (n, k) => {
  let ret = 0;

  let second = 59;
  second += 59 * 60;
  second += n * 60 * 60;

  for (let i=0; i<=second; i++)
    if(check_sec(i, k)) ret ++;
  
  return ret;
}

console.log(answer(n, k));

File: 백준/Bronze/18312. 시각/시각.cc
#include <bits/stdc++.h>
using namespace std;

int n, k, ret;

string to_string2(int a) {
  string s = "";
  if (a / 10 == 0) {
    s = '0' + to_string(a);
  } else {
    s = to_string(a);
  }
  return s;
}

bool check_string(int sec) {
  int h = 0, m = 0;

  h = sec / (60 * 60);
  sec %= 60 * 60;
  m = sec / (60);
  sec %= 60;
  string sh, sm, ss;

  sh = to_string2(h);
  sm = to_string2(m);
  ss = to_string2(sec);

  // cout << sh << ' ' << sm << ' ' << ss << '\n';
  // cout << to_string(h).find(char(k)) << '\n';
  string temp = to_string(k);
  if (sh.find(temp) != string::npos || sm.find(temp) != string::npos ||
      ss.find(temp) != string::npos) {
    // cout << "hit\n";
    return true;
  } else
    return false;
}

int main() {
  cin >> n >> k;

  int sec = 59;
  sec += 59 * 60;
  sec += n * 60 * 60;

  for (int i = sec; i >= 0; i--)
    if (check_string(i)) ret++;

  cout << ret;
  return 0;
}
